### PR TITLE
Third feature

### PR DESCRIPTION
**Pull request by sjdev2212**
In this feature I :  
Created Classroom, Books and Rentals classes
Added all the requested relationships and attributes.
Checked linters

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label)
+    @label = label
+    @student = []
+  end
+
+  def add_student(student)
+    @student.push(student)
+    student.class_room = self
+  end
```
Comment (mikethreels) [inline classroom.rb]: - [x] in order to make this work you will need to set up the attribute reader for `@students` as well

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(age, name = 'Unknown', parent_permission = 'unknown')
+    super(age, name, parent_permission)
+  end
+
+  attr_accessor :class_room
+
+  def play_hooky
+    '¯\(ツ)/¯'
+  end
+end
```
Comment (mikethreels) [inline student.rb]: - [x] in order to make this work you will need to set up the 'belongs-to' side as well as mentioned in [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+  attr_accessor :title, :author, :rentals
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (mikethreels) [inline book.rb]: - [x] in order to make this work you will need to initialize the rental class in this method as explained in [this article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`).

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Rental
+  def initialize(date)
+    @date = date
+  end
```
Comment (mikethreels) [inline rental.rb]: - [x] you're going to need to initialize the `book` and `person` as well in this method which will take both as arguments. and the rental should add itself to the instance variable created, which will look something like below:
```
    @book = book
    book.rentals << self

    @person = person
    person.rentals << self
```

File: person.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  attr_accessor :name, :age
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (mikethreels) [inline person.rb]: - [x] in order to make this work you will need to initialize the rental class in this method as explained in [this article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`).

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Anibal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require './person'
+require './classroom'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, name = 'Unknown', parent_permission = 'unknown')
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [x] what I meant with the previous review is that there needs to be a method which looks like the below example:
```
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end

```

File: book.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
```
Comment (mikethreels) [inline book.rb]: - [x] with the previous comment I meant that the rental class should be initialized similarly to below:
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    @rentals = []
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
```
Comment (mikethreels) [inline person.rb]: - [x] with the previous comment I meant that the rental class should be initialized similarly to below:
```
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Anibal, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
Hi @sj1978, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**